core/src/main/java/com/scalar/db/io/BlobValue.java

```diff
@@ -50,18 +47,15 @@ public BlobValue(byte[] value) {
     this(ANONYMOUS, value);
   }
 
-  /**
-   * Returns the content of this {@code Value}
-   *
-   * @return an {@code Optional} of the content of this {@code Value}
-   */
+  @Override
+  @Nonnull
   public Optional<byte[]> get() {
-    return bytes.map(b -> Arrays.copyOf(b.array(), b.limit()));
+    return value;
```

Comment:
Does this break the immutability? (value can be updated by the outside of the class?)